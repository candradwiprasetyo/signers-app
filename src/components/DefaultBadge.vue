<template>
  <span
    :class="[
      'material-icons text-white text-xs rounded-full w-6 h-6 flex items-center justify-center mx-auto',
      isDefault ? 'bg-green-400' : 'bg-red-400',
    ]"
    :title="isDefault ? 'Aktif' : 'Tidak Aktif'"
  >
    {{ isDefault ? 'check' : 'close' }}
  </span>
</template>

<script setup lang="ts">
defineProps({
  isDefault: Boolean,
})
</script>
