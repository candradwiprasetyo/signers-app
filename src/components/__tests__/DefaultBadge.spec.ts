import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import DefaultBadge from '@/components/DefaultBadge.vue'

describe('DefaultBadge', () => {
  it('render active icon when isDefault is true', () => {
    const wrapper = mount(DefaultBadge, {
      props: {
        isDefault: true,
      },
    })

    expect(wrapper.text()).toBe('check')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('render non active icon when isDefault is false', () => {
    const wrapper = mount(DefaultBadge, {
      props: {
        isDefault: false,
      },
    })

    expect(wrapper.text()).toBe('close')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
