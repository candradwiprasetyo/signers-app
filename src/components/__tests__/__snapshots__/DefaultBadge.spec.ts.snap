// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DefaultBadge > render active icon when isDefault is true 1`] = `"<span class="material-icons text-white text-xs rounded-full w-6 h-6 flex items-center justify-center mx-auto bg-green-400" title="Aktif">check</span>"`;

exports[`DefaultBadge > render non active icon when isDefault is false 1`] = `"<span class="material-icons text-white text-xs rounded-full w-6 h-6 flex items-center justify-center mx-auto bg-red-400" title="Tidak Aktif">close</span>"`;
